libraries/chain/controller.cpp

```diff
@@ -1592,8 +1607,10 @@ struct controller_impl {
                trace->receipt = r;
```

Comment:
I feel like we should move the assignment of `r.status`, `r.cpu_usage`, `r.net_usage_words` out of `push_receipt` and do it here for all trx types. It is odd to do it here for some trx types and in `push_receipt` for other types.

Reply:
`push_receipt` is used in several places. Many of them do not have two cases here. They just set the status, cpy_usage, and net_usage_words inside push_receipt.